Mathlib/FieldTheory/Galois/Infinite.lean

```diff
@@ -0,0 +1,207 @@
+  val_injective.lattice _ (fun _ _ ↦ rfl) (fun _ _ ↦ rfl)
+
+instance : OrderBot (FiniteGaloisIntermediateField k K) where
+  bot := .mk ⊥
+  bot_le _ := bot_le (α := IntermediateField _ _)
+
+@[simp]
+lemma le_iff (L₁ L₂ : FiniteGaloisIntermediateField k K) :
+    L₁ ≤ L₂ ↔ L₁.toIntermediateField ≤ L₂.toIntermediateField :=
+  Iff.rfl
+
+variable (k) in
+/-- The minimal (finite) Galois intermediate field containing a finite set `s : Set K` in a
+Galois extension `K/k` defined as the the normal closure of the field obtained by adjoining
+the set `s : Set K` to `k`. -/
+noncomputable def adjoin [IsGalois k K] (s : Set K) [Finite s] :
+    FiniteGaloisIntermediateField k K := {
+  normalClosure k (IntermediateField.adjoin k (s : Set K)) K with
+  finiteDimensional :=
+    letI : FiniteDimensional k (IntermediateField.adjoin k (s : Set K)) :=
+      IntermediateField.finiteDimensional_adjoin <| fun z _ =>
+        IsAlgebraic.isIntegral (Algebra.IsAlgebraic.isAlgebraic z)
+    normalClosure.is_finiteDimensional k (IntermediateField.adjoin k (s : Set K)) K
+  isGalois := IsGalois.normalClosure k (IntermediateField.adjoin k (s : Set K)) K }
+
+@[simp]
+lemma adjoin_val [IsGalois k K] (s : Set K) [Finite s] :
+    (FiniteGaloisIntermediateField.adjoin k s) =
+    normalClosure k (IntermediateField.adjoin k s) K :=
+  rfl
+
+variable (k) in
+lemma subset_adjoin [IsGalois k K] (s : Set K) [Finite s] :
+    s ⊆ (adjoin k s).toIntermediateField :=
+  (IntermediateField.subset_adjoin k s).trans (IntermediateField.le_normalClosure _)
+
+theorem adjoin_simple_le_iff [IsGalois k K] {x : K} {L : FiniteGaloisIntermediateField k K} :
+    adjoin k {x} ≤ L ↔ x ∈ L.toIntermediateField := by
+  simp only [le_iff, adjoin_val, IntermediateField.normalClosure_le_iff_of_normal,
+    IntermediateField.adjoin_le_iff, Set.le_eq_subset, Set.singleton_subset_iff, SetLike.mem_coe]
+
+@[simp]
+theorem adjoin_map [IsGalois k K] (f : K →ₐ[k] K) (s : Set K) [Finite s] :
+    adjoin k (f '' s) = adjoin k s := by
+  apply val_injective; dsimp [adjoin_val]
+  rw [← IntermediateField.adjoin_map, IntermediateField.normalClosure_map_eq]
+
+@[simp]
+theorem adjoin_simple_map_algHom [IsGalois k K] (f : K →ₐ[k] K) (x : K) :
+    adjoin k {f x} = adjoin k {x} := by
+  simpa only [Set.image_singleton] using adjoin_map f { x }
+
+@[simp]
+theorem adjoin_simple_map_algEquiv [IsGalois k K] (f : K ≃ₐ[k] K) (x : K) :
+    adjoin k {f x} = adjoin k {x} :=
+  adjoin_simple_map_algHom (f : K →ₐ[k] K) x
+
+/-- The (finite) Galois group `Gal(K / k)` associated to a
+`L : FiniteGaloisIntermediateField k K` `L`. -/
+def finGaloisGroup (L : FiniteGaloisIntermediateField k K) : FiniteGrp :=
+  letI := AlgEquiv.fintype k L
+  FiniteGrp.of <| L ≃ₐ[k] L
+
+/-- For `FiniteGaloisIntermediateField` s `L₁` and `L₂` with `L₂ ≤ L₁`
+  the restriction homomorphism from `Gal(L₁/k)` to `Gal(L₂/k)` -/
+noncomputable def finGaloisGroupMap {L₁ L₂ : (FiniteGaloisIntermediateField k K)ᵒᵖ}
```

Comment:
This map exists only as (categorical-theoretical) morphism but not as hones ring map. Can you add this, and possibly use it to build this morphism? Also, can you please move all results expressed in categorial terms to a separate section `Category` (or something like this)?

Reply:
The results expressed in categorial terms are moved to a new section currently called `profinite`.
However, what do you mean by "hones ring map"? I think I know nothing about it.

Reply:
I am sorry, I meant as _honest_ ring map, as in [RingHom](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Ring/Hom/Defs.html#RingHom)

Reply:
I think the map itself is alredy there, I just used `AlgEquiv.restrictNormalHom` . So do you mean by giving the map when the two extensions have a `RingHom`?

Reply:
Yes, you're right. I have tried and it is simply that. All in all, I think that now this PR is ready to go. Thank you very much for your patience, I hope you did not take my comments as being too pushy, but sometimes the early life of mathlib contributors is a bit hard, I remember it very well... But your contributions are really valuable.